Emails: Add sign up step - initial design

AllTerrainDeveloper opened the pull request:
#### Changes proposed in this Pull Request

This PR creates a new sign up step to allow users to add a free trial of Professional Email to their shopping cart when they select a domain during the domains step.

At present, the step is being added to the main `onboarding` flow behind the `signup/professional-email-step` feature flag.

For now, this PR _only_ adds the visuals and minimal wiring for the new step - we will work on wiring the component into more of the flows in subsequent PRs. This PR also doesn't handle the case where the user selects a free WordPress.com subdomain in the domains step.

#### Testing instructions

* Open this branch locally or via calypso.live (the current live link should be in [this comment](https://github.com/Automattic/wp-calypso/pull/54641#issuecomment-881577562))
* Navigate to `/start/onboarding`
* Work through the signup flow, making sure to select a domain registration on the domains step
* Verify that you _don't_ see the email step after selecting a domain registration on the domains step of the flow.
* Navigate to `/start`
* Work through the signup flow, making sure to select a domain registration on the domains step
* Verify that you _don't_ see the email step after selecting a domain registration on the domains step of the flow.
* Navigate to `/start/onboarding?flags=signup/professional-email-step`
* Work through the signup flow and make sure you select a domain registration on the domains step
* Verify that you see the new Professional Email step and that it looks good at various screen sizes and is visually consistent with the previous step in the flow. (Don't test the Add or Skip buttons, as they are currently not wired in.
* Navigate to `/start?flags=signup/professional-email-step`
* Work through the signup flow and make sure you select a domain registration on the domains step
* Verify that you see the new Professional Email step and that it looks good at various screen sizes and is visually consistent with the previous step in the flow. (Don't test the Add or Skip buttons, as they are currently not wired in.

#### Screenshot

<img width="1666" alt="New Professional Email step" src="https://user-images.githubusercontent.com/3376401/126254419-3428a893-6444-4c39-877f-ef106974c410.png">

Original design discussion is in pcDL0W-nj-p2.

AllTerrainDeveloper committed:
Emails: Add sign up step - initial commit

a8ci18n commented on client/signup/steps/emails/index.jsx:
```diff
@@ -0,0 +1,106 @@
+	render() {
+		const { flowName, translate, stepName, positionInFlow } = this.props;
+		const backUrl = '/emails/';
+		const headerText = translate( 'Add Professional Email' );
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **31** times:
`translate( 'Professional email address' )` *ES Score: 6*

daledupreez commented on client/components/emails/email-product-price/index.jsx:
```diff
@@ -0,0 +1,75 @@
+		const threeMonths = 86400 * 90 * 10000;
+		const startDate = Date.now() + threeMonths;
+
+		const priceText = this.props.translate( '%(cost)s per month starting %(starts)s', {
+			args: { cost: this.props.price, starts: startDate },
+		} );
```
I think this should be "per mailbox per month" for clarity, and the substitution variable names should be a bit more specific. It is _always_ good to leave a comment for the translators as well.
```suggestion
		const { price, translate } = this.props;

		const priceText = this.props.translate( '%(price)s per user per month starting %(startDate)s', {
			args: {
				price,
				startDate,
			},
			comment: '%(price)s is the price of a subscription, and %(startDate)s is the date the subscription will first renew',
		} );
```

AllTerrainDeveloper replied:
Thank you for specifying that comment. I thought you meant to add a comment in the code, not in the actual translation! 🥰

daledupreez commented on client/components/emails/email-product-price/index.jsx:
```diff
@@ -0,0 +1,75 @@
+
+	renderContentPrice() {
+		const threeMonths = 86400 * 90 * 10000;
+		const startDate = Date.now() + threeMonths;
```
It will likely be much better to use our `localizedMoment` wrapper around `moment` here, for both the addition and the localization/formatting.
```suggestion
		const startDate = moment().add( 3, 'months' ).format( 'L' );
```

daledupreez commented on client/components/emails/email-product-price/index.jsx:
```diff
@@ -0,0 +1,75 @@
+		return <div className="email-product-price__price">{ priceText }</div>;
+	}
+
+	renderContent() {
+		const className = classnames( 'email-product-price', 'is-free-domain', {
+			'email-product-price__email-step-signup-flow': this.props.showStrikedOutPrice,
+		} );
+		return (
+			<div className={ className }>
+				{ this.renderContentText() }
+				{ this.renderContentPrice() }
+			</div>
+		);
+	}
+
+	render() {
+		if ( this.props.isLoading ) {
+			return (
+				<div className="email-product-price is-placeholder">
+					{ this.props.translate( 'Loading…' ) }
+				</div>
+			);
+		}
+
+		return this.renderContent();
+	}
```
I am not sure if this is preference comes from me having worked on functional components a lot recently, but all the functions in this component make it harder for me to read. I think we could push all of the functions in this class into a single `render()` function without making it very complicated, as most of the functions are just setting up one or two `const` values and then returning JSX.

AllTerrainDeveloper replied:
I've converted this class to a funcitonal react component.

daledupreez commented on client/components/emails/email-product-price/style.scss:
```diff
@@ -0,0 +1,119 @@
+		color: var( --color-text-subtle );
+	}
+
+	.email-product-price__premium-text {
```
Is this actually specified anywhere?

daledupreez commented on client/components/emails/emails-side-explainer/checkmark.jsx:
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import { SVG, Path } from '@wordpress/components';
+
+const checkmark = (
```
Would it be simpler to just use a `.svg` file with the relevant content? Also, did you check to see if we have this checkmark or a very similar one available to us via the `Gridicon` or `MaterialIcon` components?

AllTerrainDeveloper replied:
Thank you, I didn't know we had to use GridIcons :)

daledupreez commented on client/components/emails/emails-side-explainer/index.jsx:
```diff
@@ -0,0 +1,62 @@
+ *  Internal dependencies
+ */
+import checkmark from './checkmark';
+import { Icon } from '@wordpress/icons';
```
This is technically an external dependency. 😁

daledupreez commented on client/components/emails/emails-side-explainer/index.jsx:
```diff
@@ -0,0 +1,62 @@
+ */
+import './style.scss';
+
+class EmailSideExplainer extends React.Component {
```
Minor comment: It may be worth including `EmailStep` in the component name to make its purpose a bit clearer. Something like `EmailStepSideBar` would help me understand the purpose more easily.

daledupreez commented on client/components/emails/emails-side-explainer/index.jsx:
```diff
@@ -0,0 +1,62 @@
+import './style.scss';
+
+class EmailSideExplainer extends React.Component {
+	getStrings() {
```
Nit: this function doesn't return strings!

daledupreez commented on client/components/emails/emails-side-explainer/index.jsx:
```diff
@@ -0,0 +1,62 @@
+		const { title, subtitle } = this.getStrings();
+
+		return (
+			/* eslint-disable jsx-a11y/click-events-have-key-events */
```
Why do we need these linter comments? Do you know what components are adding click handlers?

daledupreez commented on client/components/emails/emails-side-explainer/style.scss:
```diff
@@ -0,0 +1,38 @@
+        padding-top: 12px;
+        font-size: 0.875rem;
+		padding-left: 30px;
+		text-indent:-30px;
```
Can you check the indentation here? It looks like there is a mismatch.

Also, I have a strong preference for keeping the items in alphabetical order. 😳

daledupreez commented on client/components/emails/emails-titan-card/index.jsx:
```diff
@@ -0,0 +1,155 @@
+ * Style dependencies
+ */
+import './style.scss';
+import { Icon } from '@wordpress/icons';
```
This is an external dependency.

daledupreez commented on client/components/emails/emails-titan-card/index.jsx:
```diff
@@ -0,0 +1,155 @@
+ * Internal dependencies
+ */
+import EmailProductPrice from 'calypso/components/emails/email-product-price';
+import { Button } from '@automattic/components';
```
This is an external dependency.

daledupreez commented on client/components/emails/emails-titan-card/index.jsx:
```diff
@@ -0,0 +1,155 @@
+import classNames from 'classnames';
+import { localize, getLocaleSlug } from 'i18n-calypso';
+import Gridicon from 'calypso/components/gridicon';
+import moment from 'moment';
```
We have a localizedMoment wrapper via `withLocalizedMoment()` in `client/components/localized-moment/index.js` - that is preferred to using `moment` directly.

daledupreez commented on client/components/emails/emails-titan-card/index.jsx:
```diff
@@ -0,0 +1,155 @@
+ */
+import EmailProductPrice from 'calypso/components/emails/email-product-price';
+import { Button } from '@automattic/components';
+import tip from '../../domains/register-domain-step/tip';
```
It may be worth pushing this into a new location. Also, it is generally better to use the path from the root rather than relative paths like `../..`, as relative paths can be harder to refactor.

daledupreez commented on client/components/emails/emails-titan-card/index.jsx:
```diff
@@ -0,0 +1,155 @@
+						<div className="email-registration-suggestion__email-title">
+							<span className="email-registration-suggestion__email-title-name">
+								{ ' ' }
+								{ this.props.translate( 'youremail' ) }@{ siteUrl }
```
I was thinking we should translate the full string and include a comment.
```suggestion
								{ this.props.translate( 'youremail@%(domainName)s', { args: { domainName: siteUrl }, comment: 'This is a sample email address for the user at their domain; %(domainName)s is a domain name, e.g. example.com' } ) }
```

daledupreez commented on client/components/emails/emails-titan-card/index.jsx:
```diff
@@ -0,0 +1,155 @@
+					<h3 className="email-registration-suggestion__title">
+						<div className="email-registration-suggestion__email-title">
+							<span className="email-registration-suggestion__email-title-name">
+								{ ' ' }
```
Why the leading whitespace on this line? If it's needed, please add a comment to explain why.

AllTerrainDeveloper replied:
A reminder of old code. sorry. Removed.

daledupreez commented on client/components/emails/emails-titan-card/index.jsx:
```diff
@@ -0,0 +1,155 @@
+				</div>
+				<div className="email-product-price is-free-email email-product-price__email-step-signup-flow">
+					<div className="email-product-price__free-text">
+						<span className="email-product-price__free-price">Free for the first three months</span>
```
Should we be using the price component here? This text is not translated, hence the question.

AllTerrainDeveloper replied:
We are now using a react component for it.

daledupreez commented on client/components/emails/emails-titan-card/index.jsx:
```diff
@@ -0,0 +1,155 @@
+						<span className="email-product-price__free-price">Free for the first three months</span>
+					</div>
+					<div className="email-product-price__price">
+						<span>$3.5/per month starting { momentTime.format( 'LL' ) }</span>
```
We're going to need to get the localized price from the product list. (I'll look up how we ought to do that from signup this morning.)

AllTerrainDeveloper replied:
I was actually wondering how to "translate" to different prices. Thank you.

daledupreez commented on client/components/emails/emails-titan-card/index.jsx:
```diff
@@ -0,0 +1,155 @@
+			);
+
+		/* eslint-disable jsx-a11y/click-events-have-key-events */
+		/* eslint-disable jsx-a11y/interactive-supports-focus */
```
These linter comments raise red flags for me. If they're needed, I strongly prefer that they are for specific lines and the reason is well-described.

daledupreez commented on client/components/emails/emails-titan-card/index.jsx:
```diff
@@ -0,0 +1,155 @@
+	}
+}
+
+EmailTitanCard.Placeholder = function () {
```
Might it be worth pushing this to a prop on the main class so we can push the logic into this component rather than externalising it onto the consumer?

daledupreez commented on client/components/emails/emails-titan-card/test/index.js:
```diff
@@ -0,0 +1,35 @@
+/**
```
This looks like a copy-paste file. Please update it to be relevant or just delete it. 😁

AllTerrainDeveloper replied:
Deleted :)

daledupreez commented on client/components/emails/emails-titan-card/style.scss:
```diff
@@ -0,0 +1,434 @@
+@import '@wordpress/base-styles/breakpoints';
+@import '@wordpress/base-styles/mixins';
+
+.email-suggestion:not( .featured-email-suggestion ) {
```
I don't understand why we need this? Is this lingering from some original content that you started from?
If it is lingering, it would be well worth removing a lot of it so we only have applicable styles and no orphaned content.

daledupreez commented on client/signup/config/flows-pure.js:
```diff
@@ -112,7 +112,7 @@ export function generateFlows( {
 		{
 			name: 'onboarding',
-			steps: [ 'user', 'domains', 'plans' ],
+			steps: [ 'user', 'domains', 'emails', 'plans' ],
```
I mentioned this verbally yesterday, but I think we should feature flag this ASAP. I've suggested a flag name below, but we don't have to use this one.

For now, let's keep the feature disabled in all environments until we have it sufficiently wired-in/working. That means we shouldn't add it to `config/development.json` quite yet.

```suggestion
			import { isEnabled } from '@automattic/calypso-config'; // add at top of file

			steps: isEnabled( 'signup/professional-email-step' ) ? [ 'user', 'domains', 'emails', 'plans' ] : [ 'user', 'domains', 'plans' ],
```

AllTerrainDeveloper replied:
Thank you, I must admit that my intention was to keep working on it, so I delayed it, like a few things more in the code that are now solved.

daledupreez commented on client/signup/config/steps-pure.js:
```diff
@@ -32,6 +32,7 @@ export function generateSteps( {
 	createWpForTeamsSite = noop,
 	createSiteOrDomain = noop,
 	createSiteWithCart = noop,
+	buyEmailWithCart = noop,
```
Maybe this should be `addProfessionalEmailWithCart`? Or `addEmailToCart` to match `addDomainToCart`?

daledupreez commented on client/signup/config/steps-pure.js:
```diff
@@ -296,14 +297,30 @@ export function generateSteps( {
-
+		emails: {
+			stepName: 'emails',
+			dependencies: [ 'siteUrl', 'domainItem' ],
```
Do we need `siteUrl`? Shouldn't we be depending on `domainItem`? From a logical perspective, we use the `domainItem` to modify the `siteUrl`, but I think we should use `domainItem` wherever possible, as we could have situations where `domainItem` is empty but `siteUrl` is set to something like `*.wordpress.com` (i.e. a free site URL).

It may just be simpler to extract the value as `customDomainName` or `customDomain`.

daledupreez commented on client/signup/steps/emails/index.jsx:
```diff
@@ -0,0 +1,106 @@
+ * Style dependencies
+ */
+import './style.scss';
+import CalypsoShoppingCartProvider from 'calypso/my-sites/checkout/calypso-shopping-cart-provider';
```
Nit: this is an internal dependency.

daledupreez commented on client/signup/steps/emails/index.jsx:
```diff
@@ -0,0 +1,106 @@
+		);
+	}
+
+	handleSkip = () => {};
```
Maybe add a comment or TODO here?

daledupreez commented on client/signup/steps/emails/index.jsx:
```diff
@@ -0,0 +1,106 @@
+					<EmailTitanCard
+						siteUrl={ this.props.progress.domains.siteUrl }
+						addButtonContent={ 'Add' }
+						skipButtonContent={ 'Skip' }
```
Two comments:
1. These should both be translated
2. Maybe they should be called `xButtonTitle` or `xButtonText`?

daledupreez commented on client/signup/steps/emails/index.jsx:
```diff
@@ -0,0 +1,106 @@
+		const headerText = translate( 'Add Professional Email' );
+		const url = this.props.progress.domains.siteUrl;
+		const subHeaderText = translate(
+			`Add a custom email address to start sending and receiving emails from {{b}}${ url }{{/b}} today.`,
```
Maybe just use `{{strong}}` here?

daledupreez commented on client/components/emails/emails-titan-card/index.jsx:
```diff
@@ -0,0 +1,155 @@
+import './style.scss';
+import { Icon } from '@wordpress/icons';
+
+class EmailTitanCard extends React.Component {
```
This name is a bit too generic for its purpose. Maybe call this `EmailSignupTitanCard`?

daledupreez commented on client/signup/steps/emails/index.jsx:
```diff
@@ -0,0 +1,106 @@
+						skipButtonContent={ 'Skip' }
+						onAddButtonClick={ () => {} }
+						onSkipButtonClick={ () => {} }
+						price={ '3.5' }
```
Side note: This price should include the currency symbol and should be derived from the store products. (I'll share more on this after I wrap the review.)

daledupreez commented on client/signup/steps/emails/index.jsx:
```diff
@@ -0,0 +1,106 @@
+						siteUrl={ this.props.progress.domains.siteUrl }
+						addButtonContent={ 'Add' }
+						skipButtonContent={ 'Skip' }
+						onAddButtonClick={ () => {} }
+						onSkipButtonClick={ () => {} }
```
Maybe add TODO comments here?

daledupreez commented on client/signup/steps/emails/index.jsx:
```diff
@@ -0,0 +1,106 @@
+						onAddButtonClick={ () => {} }
+						onSkipButtonClick={ () => {} }
+						price={ '3.5' }
+					></EmailTitanCard>
```
1000% nit: this could be simplified slightly.
```suggestion
					/>
```

daledupreez commented on client/signup/steps/emails/index.jsx:
```diff
@@ -0,0 +1,106 @@
+class EmailsStep extends React.Component {
+	constructor( props ) {
+		super( props );
+	}
```
This looks unnecessary as it is a noop.

daledupreez commented on client/signup/steps/emails/index.jsx:
```diff
@@ -0,0 +1,106 @@
+				isExternalBackUrl={ false }
+				fallbackHeaderText={ headerText }
+				fallbackSubHeaderText={ subHeaderText }
+				stepContent={ <div>{ this.renderContent() }</div> }
```
I'd prefer this setup to occur via a `const` outside the JSX, as the wrapping div is subtle and easy to miss.

daledupreez commented on client/signup/steps/emails/index.jsx:
```diff
@@ -0,0 +1,106 @@
+
+	handleSkip = () => {};
+
+	render() {
```
Stylistic note: I find the jumping across the three rendering functions harder to follow, especially because there is very little logic in each function. They each set up some variables and then return JSX. As I noted elsewhere, I prefer those pieces to be sequential in the code, even if it does increase the function size a bit, because I can then see what pieces are going in to the returned JSX fairly easily. (This is not an instruction to change, BTW, more a comment on how I am noticing my preference! We should discuss before you make changes! )

daledupreez commented on client/signup/steps/emails/index.jsx:
```diff
@@ -0,0 +1,106 @@
+
+	render() {
+		const { flowName, translate, stepName, positionInFlow } = this.props;
+		const backUrl = '/emails/';
```
This feels like it shouldn't be hard-coded -- can you add a comment as to why this is here and/or what cleanup needs to happen?

AllTerrainDeveloper replied:
Hello! I felt that there were many backUrls to go back. I agree that this should be dynamic, as any other step does. But 
for this being dynamic, we need to know which step/workflow we are to easily go back to. 

I also think that this might be a good improvement for the steps component, it should be selfaware of what's the previous step or where this step is coming from, as it knows where to go for the next step. :)

daledupreez reviewed (commented):
Lots of comments from me. I haven't had a chance to test, but will definitely do so later.

AllTerrainDeveloper commented:
> Lots of comments from me. I haven't had a chance to test, but will definitely do so later.

I completely agree with all your comments :) I’ll update it after lunch

AllTerrainDeveloper committed:
Emails: Cleaned up some code, added Suggestions by Dale.

daledupreez commented on client/components/emails/email-product-price/index.jsx:
```diff
@@ -0,0 +1,45 @@
+import './style.scss';
+
+const EmailProductPrice = ( { isLoading, price, translate } ) => {
+	const startDate = moment().add( 3, 'months' ).format( 'LL' );
```
Can you please use `useLocalizedMoment` from `calypso/components/localized-moment` instead? That ensures we're using the correct locale for moment.

daledupreez replied:
It is also simpler to use `useTranslate()` from `i18n-calypso` inside functional components. That way you don't have to push `translate` via the props.

daledupreez commented on client/components/emails/email-signup-titan-card/index.jsx:
```diff
@@ -0,0 +1,129 @@
+		price: PropTypes.string,
+		email: PropTypes.string,
+		hidePrice: PropTypes.bool,
+		showChevron: PropTypes.bool,
```
It may be worth keeping these in alphabetical order -- this list is getting long!

daledupreez commented on client/components/emails/email-signup-titan-card/index.jsx:
```diff
@@ -0,0 +1,129 @@
+
+		return (
+			<>
+				<div
```
I see we're using the standard `Card` class names in `classes`, but we're using a `div` here. Could you explain why? (Best would be a comment in the code.)

daledupreez commented on client/components/emails/email-signup-titan-card/index.jsx:
```diff
@@ -0,0 +1,129 @@
+							</span>
+						</div>
+					</h3>
+				</div>
```
Why so many nested elements?!?! We should try to reduce this by at least one layer.

daledupreez commented on client/components/emails/email-signup-titan-card/index.jsx:
```diff
@@ -0,0 +1,129 @@
+		);
+	}
+
+	renderEmailSuggestion( siteUrl ) {
```
I've mentioned this elsewhere, but I would strongly prefer we call this `domainName` or `customDomainName`, as we definitely don't want free site URLs in this component.

daledupreez commented on client/components/emails/email-step-side-bar/index.jsx:
```diff
@@ -0,0 +1,56 @@
+ * Style dependencies
+ */
+import './style.scss';
+import Gridicon from "calypso/components/gridicon";
```
Nit: this is an internal dependency, and should be wrapped in single quotes.

daledupreez commented on client/components/emails/email-step-side-bar/index.jsx:
```diff
@@ -0,0 +1,56 @@
+import './style.scss';
+import Gridicon from "calypso/components/gridicon";
+
+class EmailStepSideBar extends React.Component {
```
Given that this component is just a basic content render, it may be simpler to write it as a functional component with a single JSX block, with `useTranslate` from `i18-calypso` at the top to handle the translation.

daledupreez commented on client/components/emails/email-step-side-bar/index.jsx:
```diff
@@ -0,0 +1,56 @@
+			<>
+				<p>
+					<Gridicon icon="checkmark" size={ 16 } />
+					{ `${ translate( 'Stand out with every email you send' ) }` }
+				</p>
+				<p>
+					<Gridicon icon="checkmark" size={ 16 } />
+					{ `${ translate( 'Single dashboard for your email, domain and website' ) }` }
+				</p>
+				<p>
+					<Gridicon icon="checkmark" size={ 16 } />
+					{ `${ translate( 'Easy to import your existing emails and contacts' ) }` }
```
I don't think we need the wrapped/nested construction here - `translate` returns a `TranslateResult` type (which you can look up) -- it's either a string or a valid React fragment, so can be embedded directly and safely.
```
{ `${ translate( ... ) }` } -> { translate( ... ) }
```

daledupreez commented on client/components/emails/email-step-side-bar/index.jsx:
```diff
@@ -0,0 +1,56 @@
+		return (
+			<div className="email-side-explainer">
+				<div className="email-side-explainer__title">{ title }</div>
+				<div className="email-side-explainer__subtitle">{ subtitle }</div>
```
I don't think "subtitle" is a good description for this -- I think calling it "content" would be more accurate.

daledupreez commented on client/components/emails/email-step-side-bar/index.jsx:
```diff
@@ -0,0 +1,56 @@
+ */
+import React from 'react';
+import { localize } from 'i18n-calypso';
+import { Icon } from '@wordpress/icons';
```
I don't think we're using `Icon` here any more.

daledupreez commented on client/components/emails/email-step-side-bar/index.jsx:
```diff
@@ -0,0 +1,56 @@
+		const { title, subtitle } = this.getExplainers();
+
+		return (
+			<div className="email-side-explainer">
```
Please update the class names to match the class, both in the JSX here and in the SCSS file.

daledupreez commented on client/landing/gutenboarding/hooks/use-steps.ts:
```diff
@@ -37,6 +37,7 @@ export default function useSteps(): Array< StepType > {
 		steps = [
 			Step.IntentGathering,
 			Step.Domains,
+			Step.Emails,
```
I don't think we should change this file -- I don't believe this code is used by our current onboarding flow. (Feel free to correct me, but this doesn't feel like it's needed.)

daledupreez commented on client/signup/steps/emails/index.jsx:
```diff
@@ -0,0 +1,104 @@
+			<div className="register-email-step">
+				<CalypsoShoppingCartProvider>
+					<EmailSignupTitanCard
+						siteUrl={ this.props.progress.domains.siteUrl }
```
I think we should be pulling this from `domainItem` rather than `domains.siteUrl`.

AllTerrainDeveloper replied:
Thanks to this I also understood how to connect the component to the sign up component and take the dependencies from other steps, thanks.

daledupreez commented on client/signup/steps/emails/index.jsx:
```diff
@@ -0,0 +1,104 @@
+						onAddButtonClick={ () => {} }
+						onSkipButtonClick={ () => {} }
+						//TODO
+						price={ '3.5$' }
```
I think we should be using the `getProductDisplayCost` selector with a product slug of `TITAN_MAIL_MONTHLY_SLUG`.

We may eventually switch to the `getProductCost` selector and then passing that through `formatCurrency`, but that is not needed for this first iteration.

daledupreez reviewed (commented):
Thanks for the quick responses! I think this is looking good. I've done some basic testing, but haven't had enough time for detailed exploration of the CSS and HTML structure yet.

Also, it may be worth pointing out in the testing instructions how to access the visuals, especially seeing as we've now added a feature flag!

I was able to see the content by navigating to `/start/onboarding?flags=signup/professional-email-step` and then selecting a custom domain in the domains step. (The step didn't appear without the feature flag enabled, which is also really important!)

AllTerrainDeveloper committed:
Emails: PR Suggestions
Convert sidebar explainer into functional component
Rename a few CSS classes
Correctly connect the step with the workflow dependencies
Don't use variables inside translations

daledupreez commented on client/components/emails/email-product-price/style.scss:
```diff
@@ -50,14 +53,25 @@
+		font-width: normal;
+
+		@include break-mobile {
+			font-weight: 500; /* stylelint-disable-line */
```
I think we should not be disabling the style here, but use one of our standard font weights, which should probably 600 in this case.

daledupreez replied:
Update: we're using 500 elsewhere in signup. No need to change this for now.

daledupreez commented on client/components/emails/email-signup-titan-card/index.jsx:
```diff
@@ -0,0 +1,140 @@
+		return (
+			<div className="email-signup-titan-card__title-wrapper">
+				<h3 className="email-signup-titan-card__title">
+					<div className="email-signup-titan-card__email-title email-signup-titan-card__email-title-name">
```
This still feels unnecessarily nested to me. But we can untangle that in a follow-up PR.

daledupreez commented on client/components/emails/email-signup-titan-card/index.jsx:
```diff
@@ -0,0 +1,140 @@
+							<Button className="email-signup-titan-card__suggestion-action">
+								{ this.props.skipButtonTitle }
+							</Button>
+							<Button className="email-signup-titan-card__suggestion-action is-primary">
```
Nit: the primary handling is normally done via the explicit `isPrimary` prop on the `Button` component.
```suggestion
							<Button className="email-signup-titan-card__suggestion-action" isPrimary>
```

daledupreez replied:
My suggestion here was incorrect - the right prop is just `primary`.

daledupreez commented on client/components/emails/email-signup-titan-card/index.jsx:
```diff
@@ -0,0 +1,140 @@
+import { connect } from 'react-redux';
+import classNames from 'classnames';
+import { localize } from 'i18n-calypso';
+import Gridicon from 'calypso/components/gridicon';
```
Nit: this is technically an internal dependency.

daledupreez commented on client/components/emails/email-signup-titan-card/style.scss:
```diff
@@ -151,20 +151,20 @@
 	}
 
-	.email-registration-suggestion__hsts-tooltip {
+	.email-signup-titan-card__hsts-tooltip {
```
I don't believe we render this component -- if so, can we remove the CSS/section for it?

daledupreez commented on client/components/emails/email-signup-titan-card/style.scss:
```diff
@@ -151,20 +151,20 @@
 
 	body.is-section-signup .layout:not( .dops ) & {
-		.email-registration-suggestion__hsts-tooltip {
+		.email-signup-titan-card__hsts-tooltip {
```
Same again for this unused selector.

daledupreez commented on client/components/emails/email-signup-titan-card/style.scss:
```diff
@@ -283,84 +283,36 @@ body.is-section-signup.is-white-signup {
-			}
-		}
-
 		&:not( .featured-email-suggestion ) {
```
Without looking closely, do we ever show this branch? Or does it always apply? I think we can simplify the CSS either way, as I don't believe we have two different styles in place, do we?

daledupreez commented on client/components/emails/email-signup-titan-card/style.scss:
```diff
@@ -283,84 +283,36 @@ body.is-section-signup.is-white-signup {
 				}
 
-				.email-registration-suggestion__email-title-tld {
+				.email-signup-titan-card__email-title-tld {
```
This definitely looks like an unused CSS selector.

daledupreez commented on client/components/emails/email-step-side-bar/index.jsx:
```diff
@@ -0,0 +1,56 @@
+
+	return (
+		<div className="email-step-side-bar">
+			<div className="email-step-side-bar__title">{ title }</div>
```
For this use case, it may be simpler to just return a single block of JSX with the `translate` calls inlined.

Not blocking, but keeps the code ultra-quick to scan quickly (which suits a really simple component like this).

daledupreez commented on client/signup/steps/emails/index.jsx:
```diff
@@ -2,78 +2,80 @@
  * Internal dependencies
  */
+import CalypsoShoppingCartProvider from 'calypso/my-sites/checkout/calypso-shopping-cart-provider';
+import { getSignupDependencyStore } from 'calypso/state/signup/dependency-store/selectors';
```
Nit: this could be added after the `Email*` components to keep things in alphabetical order.

daledupreez commented on client/signup/steps/emails/index.jsx:
```diff
@@ -2,78 +2,80 @@
-						addButtonContent={ 'Add' }
-						skipButtonContent={ 'Skip' }
+					<EmailSignupTitanCard
+						siteUrl={ this.props.signupDependencies.domainItem.meta }
```
You may want (or need) a null-safe check/path here.
```suggestion
						siteUrl={ this.props.signupDependencies.domainItem?.meta }
```

daledupreez reviewed (commented):
You've wrapped for today, so I'll start work on some of these issues in the next little while.

daledupreez committed:
Small wording changes

daledupreez committed:
Reorder some CSS selectors

daledupreez committed:
Remove unused CSS selectors and styles

daledupreez committed:
Improve styles for pricing text - matches domain step

daledupreez committed:
CSS cleanup for main card

daledupreez committed:
More CSS cleanup, simplification, and re-ordering

daledupreez committed:
More CSS cleanup; simplify title HTML and CSS; group card content

daledupreez committed:
Restructure some code; update a translation; don't break when domainItem doesn't exist

daledupreez committed:
Wire in price; first step towards skip handling

daledupreez reviewed (approved):
Both @AllTerrainDeveloper and I tested this some more this morning, and I specifically retested today to make sure that "normal" signup flows are not impacted (they aren't!). So we're going to flag this as approved and will get it deployed.

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6254408

Thank you @AllTerrainDeveloper for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6254408).
